src/dataset-builder/CohortEditor.ts

```diff
@@ -610,6 +618,10 @@ export const CohortEditor: React.FC<CohortEditorProps> = (props) => {
             onClick: () => {
               updateCohorts((cohorts) => {
                 const index = _.findIndex((c) => _.equals(c.name, cohort.name), cohorts);
+                if (index === -1) {
+                  // Only add to selectedCohorts on creation of new cohort
+                  addSelectedCohort(cohort);
```

Comment:
Did you consider doing this inside `updateCohorts` instead? If the rule is we want to select any cohort that was added, this method could check the set of cohorts before and after an update and then add the new cohorts to the selection. That would avoid the need to pass down two different methods to a subcomponent, since the code that manages the selection also manages the cohort list state.

Reply:
UpdateCohorts is directly calling the setState for cohorts and is used rather broadly. We only pass addSelectedCohort() into one child method. 

Is there a straightforward way to compare the arrays and only get the new items, maybe with lodash? I'd be worried that logic would be complicated/hard to read.